compiler/src/compiler/error.rs

```diff
@@ -76,6 +80,7 @@ impl Display for CompilerErrorPayload {
                 AstError::ListWithoutClosingParenthesis => {
                     "This list doesn't have a closing parenthesis."
                 }
+                AstError::ParenthesizedInPattern => "You can not use parentheses in patterns.",
```

Comment:
For consistency with `AstError::PipeInPattern`:

```suggestion
                AstError::ParenthesizedInPattern => "Parentheses are not allowed in patterns.",
```

Maybe we should also mention that they're fine for list literals? But we could also do that once we add detailed descriptions for the error messages